tests/python/ImageClassifierTest.py

```diff
@@ -0,0 +1,230 @@
+import os
+import subprocess
+import sys
+import unittest
+
+
+class Image:
+    def __init__(self, filename, labels):
+        self.filename = filename
+        self.labels = labels
+
+
+class Model:
+    def __init__(
+        self,
+        name,
+        dataset,
+        input_name,
+        image_mode,
+        imagenet_norm=False,
+        batching=True,
+        compute_softmax=False,
+        image_layout=None,
+        label_offset=None,
+    ):
+        self.name = name
+        self.dataset = dataset
+        self.input_name = input_name
+        self.image_mode = image_mode
+        self.imagenet_norm = imagenet_norm
+        self.batching = batching
+        self.compute_softmax = compute_softmax
+        self.image_layout = image_layout
+        self.label_offset = label_offset
+
+    def command_line(self, images):
+        return (
+            ["./bin/image-classifier"]
+            + images
+            + (["-use-imagenet-normalization"] if self.imagenet_norm else [])
+            + ["-image_mode", self.image_mode]
+            + ["-m", self.name]
+            + ["-model_input_name", self.input_name]
+            + (["-compute_softmax"] if self.compute_softmax else [])
+            + (["-image_layout", self.image_layout] if self.image_layout else [])
+            + (["-label_offset", self.label_offset] if self.label_offset else [])
+        )
+
+    def check_output(self, images):
+        return subprocess.check_output(self.command_line(images)).decode("ascii")
+
+    def image_path(self, image):
+        return os.path.join("./tests/images", self.dataset, image.filename)
+
+    def images(self):
+        return [self.image_path(img) for img in images[self.dataset]]
+
+    def execute(self, testcase):
+        if self.batching:
+            out = self.check_output(self.images())
+        else:
+            out = "".join(self.check_output([image]) for image in self.images())
+        predictions = [
+            line.split()[3] for line in out.split("\n") if line.startswith(" File:")
+        ]
+        for prediction, image in zip(predictions, images[self.dataset]):
+            testcase.assertIn(int(prediction), image.labels)
```

Comment:
this is cool!

Reply:
Yeah, I was a bit torn whether to make the predicted results specific to each model.  It kind of bugs me that our networks can't decide if cat_285 is a tabby cat or Egyptian cat :-p